pkg/apis/istio/v1alpha1/istioinstaller_types.proto

```diff
@@ -0,0 +1,585 @@
+message TrafficManagementFeatureSpec {
+    // Selects whether traffic management is installed.
+    // Must be enabled to enable any sub-component.
+    google.protobuf.BoolValue enabled = 1;
+    // Namespace that all traffic management components are installed into.
+    string namespace = 2;
+
+    // Component specific config.
+    PilotComponentSpec pilot = 30;
+    ProxyComponentSpec proxy = 31;
+    SidecarInjectorComponentSpec sidecar_injector = 32;
+}
+
+// Configuration options for the policy feature.
+message PolicyFeatureSpec {
+    // Selects whether policy is installed.
+    // Must be enabled to enable any sub-component.
+    google.protobuf.BoolValue enabled = 1;
+    // Namespace that all policy components are installed into.
+    string namespace = 2;
+
+    // Default behavior of the sidecar for handling outbound traffic from the application.
+    enum OutboundPolicy {
+        // Outbound traffic to unknown destinations will be allowed, in case there are no
+        // services or ServiceEntries for the destination port
+        ALLOW_ANY = 0;
+        // Restrict outbound traffic to services defined in the service registry as well
+        // as those defined through ServiceEntries
+        REGISTRY_ONLY = 1;
+    }
+
+    // If set, allows traffic in cases when the mixer policy service cannot be reached.
+    google.protobuf.BoolValue policy_check_fail_open = 10;
+    // Selects the outbound traffic policy mode.
+    OutboundPolicy outbound_traffic_policy_mode = 11;
+
+    PolicyComponentSpec policy = 50;
+}
+
+// Configuration options for the telemetry feature.
+message TelemetryFeatureSpec {
+    // Selects whether telemetry is installed.
+    // Must be enabled to enable any sub-component.
+    google.protobuf.BoolValue enabled = 1;
+    // Namespace that all telemetry components are installed into.
+    string namespace = 2;
+
+    TelemetryComponentSpec telemetry = 51;
+}
+
+// Configuration options for security feature.
+message SecurityFeatureSpec {
+    // Selects whether security feature is installed. Must be set for any sub-component to be installed.
+    google.protobuf.BoolValue enabled = 1;
+
+    // Namespace that security components are installed into.
+    string namespace = 2;
+
+    // Enables MTLS for control plane.
+    // Will result in delays starting the pods while secrets are propagated, not recommended for tests.
+    google.protobuf.BoolValue control_plane_mtls = 13;
+
+    // Enables strict MTLS for service to service traffic, or permissive mode MTLS if unset.
+    google.protobuf.BoolValue data_plane_mtls_strict = 14;
+
+    CitadelComponentSpec citadel = 20;
+    CertManagerComponentSpec cert_manager = 21;
+    NodeAgentComponentSpec node_agent = 22;
+}
+
+// Configuration options for configuration management feature.
+message ConfigManagementFeatureSpec {
+    // Selects whether security feature is installed. Must be set for any sub-component to be installed.
```

Comment:
Nit: should be `config management`, instead of `security`

Reply:
thanks, fixed.